extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/devconsole/DevBeanInfo.java

```diff
@@ -49,42 +49,47 @@ public static DevBeanInfo from(BeanInfo bean, CompletedApplicationClassPredicate
             DevBeanKind kind;
             String memberName;
             boolean isApplicationBean;
+            boolean isGenerated = false;
             Name declaringClass;
             if (target.kind() == Kind.METHOD) {
                 MethodInfo method = target.asMethod();
                 memberName = method.name();
                 kind = DevBeanKind.METHOD;
                 isApplicationBean = predicate.test(bean.getDeclaringBean().getBeanClass());
                 declaringClass = Name.from(bean.getDeclaringBean().getBeanClass());
+                isGenerated = bean.getDeclaringBean().getImplClazz().isSynthetic();
             } else if (target.kind() == Kind.FIELD) {
                 FieldInfo field = target.asField();
                 memberName = field.name();
                 kind = DevBeanKind.FIELD;
                 isApplicationBean = predicate.test(bean.getDeclaringBean().getBeanClass());
                 declaringClass = Name.from(bean.getDeclaringBean().getBeanClass());
+                isGenerated = bean.getDeclaringBean().getImplClazz().isSynthetic();
             } else if (target.kind() == Kind.CLASS) {
                 ClassInfo clazz = target.asClass();
                 kind = DevBeanKind.CLASS;
                 memberName = null;
                 isApplicationBean = predicate.test(clazz.name());
+                isGenerated = clazz.isSynthetic();
                 declaringClass = null;
             } else {
                 throw new IllegalArgumentException("Invalid annotation target: " + target);
             }
             return new DevBeanInfo(bean.getIdentifier(), kind, isApplicationBean, providerType, memberName, types, qualifiers,
                     scope, declaringClass,
-                    interceptors);
+                    interceptors, isGenerated);
         } else {
             // Synthetic bean
             return new DevBeanInfo(bean.getIdentifier(), DevBeanKind.SYNTHETIC, false, providerType, null, types, qualifiers,
                     scope, null,
-                    interceptors);
+                    interceptors, true);
```

Comment:
Hm, synthetic beans do not require a generated type. So it might be confusing. You can test the `bean.getProviderType()` instead.

Reply:
I don't really understand this, can you elaborate?

Reply:
For synthetic beans you can use `bean.getImplClazz().isSynthetic()` instead of `true`. Because you can have a synthetic bean that provides e.g. `org.acme.Foo` and not a generated/synthetic class.

Reply:
Okay thanks